core/zoe/contracts/simpleOffer/srcs/secondPriceSrcs.js

```diff
@@ -0,0 +1,107 @@
+import harden from '@agoric/harden';
+
+import { makeHasOkRules } from '../../../contractUtils';
+
+const hasOkRulesInitialOffer = makeHasOkRules([
+  ['haveExactly', 'wantAtLeast'], // The initial offer
+]);
+
+const hasOkRulesBids = makeHasOkRules([
+  ['wantExactly', 'haveAtMost'], // Subsequent bids
+]);
+
+const isValidInitialOfferDesc = newOfferDesc =>
+  hasOkRulesInitialOffer(newOfferDesc);
+
+const isValidBid = (assays, initialOffer, newBid) =>
+  hasOkRulesBids(newBid) &&
+  assays[1].includes(newBid[1].amount, initialOffer[1].amount);
+
+const isValidOffer = (
+  assays,
+  offerIds,
+  offerIdsToOfferDescs,
+  offerMadeDesc,
+) => {
+  const isInitialOffer = offerIds.length === 0;
+  return (
+    (isInitialOffer && isValidInitialOfferDesc(offerMadeDesc)) ||
+    (!isInitialOffer &&
+      isValidBid(assays, offerIdsToOfferDescs.get(offerIds[0]), offerMadeDesc))
+  );
+};
+
+// Iterate over all of the bids keeping the highest and second highest bid.
+const findWinnerAndPrice = (strategy, offerIds, bids) => {
+  let highestBid = strategy.empty();
+  let secondHighestBid = strategy.empty();
+  let offerIdHighestBid;
+  // If the bid is greater than the highest bid, it is the new highest
+  // bid.
+  // Has side effects
+  // eslint-disable-next-line array-callback-return
+  offerIds.map((offerId, i) => {
+    const bid = bids[i];
+    if (strategy.includes(bid, highestBid)) {
+      secondHighestBid = highestBid;
+      highestBid = bid;
+      offerIdHighestBid = offerId;
+    } else if (strategy.includes(bid, secondHighestBid)) {
+      // If the bid is not greater than the highest bid, but is greater
+      // than the second highest bid, it is the new second highest bid.
+      secondHighestBid = bid;
+    }
+  });
+  return {
+    winner: offerIdHighestBid,
+    price: secondHighestBid,
+  };
+};
+
+const reallocate = (
+  strategies,
+  offerIds,
+  offerIdsToOfferDescs,
+  getQuantitiesFor,
+) => {
+  const bids = offerIds
+    .map(offerId => offerIdsToOfferDescs.get(offerId)[1].amount.quantity)
+    .slice(1);
```

Comment:
comment that this is to remove the person who created the auction. (IIUC)

Shouldn't slice() be called on offerIds rather than the result of the map() call?

Reply:
Yes